credentials/src/main/kotlin/web5/sdk/credentials/PresentationExchange.kt

```diff
@@ -31,75 +30,128 @@ public object PresentationExchange {
    *
-   * @param vcJwt The Verifiable Credential JWT as a string.
-   * @param presentationDefinition The Presentation Definition to validate against.
-   * @throws UnsupportedOperationException If the Presentation Definition's Submission Requirements
-   * feature is not implemented.
+   * This function ensures that the provided VCs meet the criteria defined in the Presentation Definition.
+   * It first checks for the presence of Submission Requirements in the definition and throws an exception if they exist,
+   * as this feature is not implemented. Then, it maps the input descriptors in the presentation definition to the
+   * corresponding VCs. If the number of mapped descriptors does not match the required count, an error is thrown.
+   *
+   * @param vcJwts List of VCs in JWT format to validate.
+   * @param presentationDefinition The Presentation Definition V2 object against which VCs are validated.
+   * @throws UnsupportedOperationException If Submission Requirements are present in the definition.
+   * @throws PresentationExchangeError If the number of input descriptors matched is less than required.
    */
   public fun satisfiesPresentationDefinition(
-    vcJwt: String,
+    vcJwts: List<String>,
     presentationDefinition: PresentationDefinitionV2
   ) {
-    val vc = JWTParser.parse(vcJwt) as SignedJWT
-
     if (!presentationDefinition.submissionRequirements.isNullOrEmpty()) {
       throw UnsupportedOperationException(
         "Presentation Definition's Submission Requirements feature is not implemented"
       )
     }
 
-    presentationDefinition.inputDescriptors
-      .filter { !it.constraints.fields.isNullOrEmpty() }
-      .forEach { inputDescriptorWithFields ->
-        validateInputDescriptorsWithFields(inputDescriptorWithFields, vc.payload)
+    val inputDescriptorToVcMap = mapInputDescriptorsToVCs(vcJwts, presentationDefinition)
+
+    if (inputDescriptorToVcMap.size != presentationDefinition.inputDescriptors.size) {
+      throw PresentationExchangeError(
+        "Missing input descriptors: The presentation definition requires " +
+          "${presentationDefinition.inputDescriptors.size} descriptors, but only " +
+          "${inputDescriptorToVcMap.size} were found. Check and provide the missing descriptors."
+      )
+    }
+  }
+
+  private fun mapInputDescriptorsToVCs(
+    vcJwtList: List<String>,
+    presentationDefinition: PresentationDefinitionV2
+  ): Map<InputDescriptorV2, List<String>> {
+    val map = mutableMapOf<InputDescriptorV2, MutableList<String>>()
+
+    presentationDefinition.inputDescriptors.forEach { inputDescriptor ->
+      val satisfyingVCs = mutableListOf<String>()
+
+      vcJwtList.forEach { vcJwt ->
+        if (vcSatisfiesInputDescriptor(vcJwt, inputDescriptor)) {
+          satisfyingVCs.add(vcJwt)
+        }
+      }
+
+      if (satisfyingVCs.size > 0) {
+        map[inputDescriptor] = satisfyingVCs
       }
+    }
+
+    return map
   }
```

Comment:
Perhaps this is simpler?
```suggestion
  private fun mapInputDescriptorsToVCs(
    vcJwtList: Iterable<String>,
    presentationDefinition: PresentationDefinitionV2
  ): Map<InputDescriptorV2, List<String>> {
    return presentationDefinition.inputDescriptors.associateWith { inputDescriptor ->
      val satisfyingVCs = vcJwtList.filter { vcJwt ->
        vcSatisfiesInputDescriptor(vcJwt, inputDescriptor)
      }
      satisfyingVCs
    }.filterValues { it.isNotEmpty() }
  }
```

Reply:
Fair enough! If you keep the old syntax, can you make sure that you aren't returning a `MutableList` object in the map? You can dynamically create immutable collections using the `buildList` pattern as shown in https://kotlinlang.org/docs/constructing-collections.html#concrete-type-constructors